### PR TITLE
allow multiple files with the same name

### LINKED ISSUE #303: Naming conflicts not handled
I'm working on an Ember-CLI project, and the naming conventions mean you have a lot of naming conflicts (multiple index.coffee etc for routes + controllers etc). As the path is not reflected in the jumpTo menu, it's impossible to distinguish between them - plus they end up overwriting each other when they get generated anyway.

If would suggest that before generating any output files, the filenames are checked for any conflicts, and if any are found, default to including the path in the jumpTo title (as has been suggestion in #252) and either name the html document differently (`controllers/index.md` --> `controllers_index.html`), or mirror the project folder structure in the `docs` folder (`controllers/index.md` --> `controllers/index.html`). The latter option would mitigate against further naming conflicts.

### PR DESCRIPTION
**Pull request by thisguychris**
- taken from the source of @karlspalding on #236 rebased with current master
- fixes #303
- closes #236
- tested on my local it works.

screenshot

### CODE REVIEW

**Comment by mendeza** [comment]
Either this or https://github.com/jashkenas/docco/pull/335

**Comment by joaomilho** [comment]
LGTM, what can we do to help it reach master?

**Comment by mercmobily** [comment]
I am just using a version of Docco patched by hand, implementing this. I don't suppose I could humbly ask you to merge either this, or #335? That would be _awesome_...!

**Comment by jashkenas** [comment]
`fs.outputFileSync`? How is that going to work?

**Comment by mercmobily** [comment]
from the documentation of [outputFileSync](https://github.com/jprichardson/node-fs-extra/blob/master/docs/outputFile.md):

> Almost the same as writeFile (i.e. it overwrites), except that if the parent directory does not exist, it's created. file must be a file path (a buffer or a file descriptor is not allowed). options are what you'd pass to fs.writeFile().

It's important since it might create a file in a sub-directory.
Docco uses `fs-extra`, hence the function is available.

File: resources/classic/docco.jst [hunk 1]
```diff
@@ -19,7 +19,7 @@
           <div id="jump_page">
             <% for (var i=0, l=sources.length; i<l; i++) { %>
               <% var source = sources[i]; %>
-              <a class="source" href="<%= path.basename(destination(source)) %>">
+              <a class="source" href="<%= relative(destination(source)) %>">
                 <%= path.basename(source) %>
```
Comment (jashkenas) [inline resources/classic/docco.jst]: You don't want to leave `path.basename(source)` in the template here, do you?

**Comment by mercmobily** [comment]
@thisguychris are you able to review/make the requested change so that this gets merged?

**Comment by jashkenas** [comment]
I've just merged in https://github.com/jashkenas/docco/commit/0deced331dd284ffbb6f28f61a17d7ea8ee3ca7a, and published it out as Docco 0.8.0. 

Let me know if it works for you!

**Pull request merged**